v0.1.0 and later by default only works in node.js (no browser support without polyfill)

This is the error logged for a user using next.js ([project link](https://github.com/PhilippeR26/Starknet-Ledger-Wallet)):
```
Import trace for requested module:
./node_modules/yauzl/index.js
./node_modules/decompress-unzip/index.js
./node_modules/decompress/index.js
./node_modules/starknet-devnet/dist/version-handler.js
./node_modules/starknet-devnet/dist/devnet.js
./node_modules/starknet-devnet/dist/index.js
./src/app/(site)/components/client/SelectNetwork.tsx
./src/app/(site)/components/client/LedgerWallet.tsx

./node_modules/graceful-fs/graceful-fs.js
Module not found: Can't resolve 'fs'
```

# 1. Polyfill solutions (temporary) - suggested by ChatGPT

## 1a. Implementing Polyfills in Next.js

Add a Custom Webpack Configuration (next.config.js):

By default, Next.js uses a custom Webpack configuration, so you can modify the configuration to include polyfills.

Add the following to your next.config.js:

```javascript
Copy code
// next.config.js
module.exports = {
  webpack: (config) => {
    // Configure fallbacks for server-only modules
    config.resolve.fallback = {
      ...config.resolve.fallback,
      fs: false, // Set `fs` to `false` to avoid bundling
      path: require.resolve('path-browserify'), // Use browser polyfill for `path`
      os: require.resolve('os-browserify/browser'), // Use browser polyfill for `os`
    };
    return config;
  },
};
```

### Explanation

`fs: false` tells Webpack to ignore the fs module in the browser environment.
path and os are replaced with browser-compatible polyfills like path-browserify and os-browserify.

## 1b. Install Browser-Compatible Polyfills:

If you want to use polyfills for specific modules, install them via npm

# Long term solution

Use a decompression library for `.tar.gz` files, not relying on `fs`. The `decompress` library, currently used, does rely on it. If `fs` issues are still present due to direct imports in `starknet-devnet-js` code, consider using dynamic imports of `fs`.

## EDIT

Also `net` and `child_process` need to be set to `false` in the webpack-config approach.

--- a/README.md
+++ b/README.md
@@ -12,9 +12,17 @@ npm i starknet-devnet
 
 # Compatibility
 
+## Devnet compatibility
+
 This library version is compatible with Devnet `v0.2.0`.
 
-[Devnet's balance checking functionality](https://0xspaceshard.github.io/starknet-devnet-rs/docs/balance#check-balance) is not provided in this library because it is simply replaceable using starknet.js, as witnessed by the [getAccountBalance function](./test/util.ts#L57)
+[Devnet's balance checking functionality](https://0xspaceshard.github.io/starknet-devnet-rs/docs/balance#check-balance) is not provided in this library because it is simply replaceable using starknet.js, as witnessed by the [getAccountBalance](./test/util.ts#L57) function.
+
+## Environment compatibility
+
+This library is intended for use with Node.js, not in a browser environment. In browsers, you can only use [`DevnetProvider`](#connect-to-a-running-instance) for connecting to an already running Devnet instance, but you cannot [spawn a new Devnet](#spawn-a-new-devnet), because that relies on modules not present in the browser engine.
+
+To enable the use of `DevnetProvider` in browser, you need to configure sources for modules otherwise reported as not found. See [this issue](https://github.com/0xSpaceShard/starknet-devnet-js/issues/26) and [this SO answer](https://stackoverflow.com/a/51669301) for more info, but generally, if using webpack, it should be enough to populate a config file with the desired polyfill implementations or with `false` values.
 
 # Usage
 

